### ISSUE TITLE
Whitespacing in edges (arrows)

### ISSUE THREAD

**Issue by gmwe** [christianvoigt/argdown#28]
Is it intentional that the output from (1)
```
[thesis1]

<arg1>
  <+ [thesis1]
  +> <arg2>

<arg2>
```
`output: thesis +> arg1 +> arg2`
and (2)
```
[thesis1]

<arg1>
  <+ [thesis1]
   +> <arg2>

<arg2>
```
`output: thesis +> arg1, thesis +> arg2`
differ?

It seems unintuitive that within an edge section for one element an edge can be made for another
element.

**Comment by christianvoigt** [comment]
Yes, this is intentional. Argdown supports defining arbitrarily complex argumentation structures within a single tree of statements and arguments. This includes circular structures. It is not based on the way graph formats like .dot define edges, but on the way stacked lists are defined in Markdown.

However, this does not mean that it is always best to define every relation of all connected nodes in one tree. Often it is better to stick with one indentation level and use only outgoing relations as you seem to prefer. 

Argdown tries to avoid being opinionated about questions of style or structure.

**Comment by gmwe** [comment]
Ok, thank you for clarification.
I stumbled upon this when using the demo editor. Creating a newline with return does not start the next line in the same level as the previous one when tabs are used. 

An example:
```
[intro]: test
	+> <a>
```
A return would lead to:
```
[intro]: test
	+> <a>
    +> <b>
``` 
(one tab vs 4 spaces, leading to test +> a +> b)

When using a 'real' editor this shouldn't be an issue since most can automatically convert tabs to spaces, but as for the demo editor i expected newlines jumping to the same (tabbed) level of intendent / replacing tabs with spaces.

**Comment by christianvoigt** [comment]
I agree, this is a problem in the demo editor.

In general, mixing tabs and spaces easily leads to errors as the indentation is determined by simply counting indentation characters, regardless if it is a tab or a space. We could count 1 indentation step for every space and 2 or 4 for every tab key, but this is also not foolproof. 

So far I haven't spend too much time on the demo editor, but adding the "convert tabs to spaces" feature seems to be easy and would fix this. It is even explained in the [manual](https://codemirror.net/doc/manual.html#keymaps):

````javascript
editor.setOption("extraKeys", {
  Tab: function(cm) {
    var spaces = Array(cm.getOption("indentUnit") + 1).join(" ");
    cm.replaceSelection(spaces);
  }
});
```

**Comment by christianvoigt** [comment]
For later reference: the relevant issue in the CodeMirror repository has more information on this. See: https://github.com/codemirror/CodeMirror/issues/988

**Comment by christianvoigt** [comment]
fixed in v1.1.0
